examples/threading/threading.c

```diff
@@ -0,0 +1,55 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+// Optional: use these functions to add debug or error prints to your application
+//#define DEBUG_LOG(msg,...)
+#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_data_p = (struct thread_data *) thread_param;
+    
+    //start with passing
+    thread_data_p->thread_complete_success = true;
+    
+    //wait for thread_data_p->wait_obtain_ms
+    usleep(thread_data_p->wait_obtain_ms); //usleep is the milisecond version of sleep
+    
+    //obtain thread_data_p->m
+    int result = pthread_mutex_lock(thread_data_p->m);
+    if(result != 0) thread_data_p->thread_complete_success = false;
```

Comment:
Upon failure, you need to return with thread params
or
Need to free thread params and return NULL